### ISSUE TITLE
don't insert inline styles, especially if  CSS_PREFIX is overwritten

### ISSUE THREAD

**Issue by algbeta** [bencripps/react-redux-grid#77]
Hi!

First of all, thank you for a nice lib!
But there is a question I struggle with.

As far as i can see there is a big amount of styles inserted directly into html as internal styles. In case I would like to override `CSS_PREFIX` i still have to handle a heavier page with those styles. 
Either I can not simply override styles (for example remove shadow from the container) in my external style file since internal styles have higher priority!

I would really appreciate if you reorganize styles as an external stylesheet.
Thanks!

**Comment by bencripps** [comment]
I think there would be some merit to including the styles as an external stylesheet, but I'm also worried about the drawbacks.

For your example, where you want to remove/override the `box-shadow`, the style is being provided by the following class:

````css
.react-grid-container {
    box-shadow: 2px 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
}
````

In your app, you could simply provide a more specific rule, where all your grid override styles are nested inside some custom class, so the more specific rule would win.

````css
.my-app .react-grid-container {
    /* override the shadow */ 
}
````

None of the grid styles are inlined into the actual grid component, so adding `!important` won't be necessary. 

The drawbacks of adding the styles as an external stylesheet would be another network request, which wouldn't be awful, but I'm not sure it's worth deconstructing the styles from the component, since you get the same results by providing a more specific rule. I've tried to keep the grid styles as flat as possible, so overriding them shouldn't be difficult.

I've taken a look around at other React components and I've noticed that they are either:

A. Inlining styles (like material-ui) with a library like Radium.
B. Including stylesheets in the HTML (like the grid).
C. Or requiring CSS via a `require` call, which puts the onus on the developer.

Of all these, I prefer B, but I am open to having the CSS load as a separate sheet for the reasons you mention. What do you think about simply using a more specific rule instead?

**Comment by algbeta** [comment]
On my point of view, the best variant is C because it's more flexible. Is it possible at least skip the styles in case i change CSS_PREFIX? Because in that case I will have unused css inserted in my html page.

Thanks!

**Comment by bencripps** [comment]
What if we did something like this:

Since we already have a construct for altering the runtime of the grid component, why not have a parameter which optionally loads the CSS. By default this will be `true`, however if you run the following function before you're grid instance, the CSS will not be loaded into the DOM.

````js

import React from 'react';
import { render } from 'react-dom';
import { Grid, Store as store, applyGridConfig } from 'react-redux-grid';

applyGridConfig({
    USE_GRID_STYLES: false
});

render(
    <Grid
        data={data}
        stateKey={stateKey}
        store={store} // or pass your existing redux store
    />,
    document.getElementById('grid-mount')
);
````

Check out `feature/css_optional` and run `npm run dist` inside the directory, and see if this works for you.

**Comment by algbeta** [comment]
that sounds fantastic! thank you a lot!

**Comment by bencripps** [comment]
Ok cool -- there are some kinks to get worked out. Once I am done, I'll publish a new minor release.

**Comment by bencripps** [comment]
Released in [5.1.0](https://github.com/bencripps/react-redux-grid/releases/tag/5.1.0)!
